tests/test_data_store_delete_operations.py

```diff
@@ -0,0 +1,213 @@
+                "test_platform_type", self.change_id
+            ).name
+            self.sensor_type = self.store.add_to_sensor_types(
+                "test_sensor_type", self.change_id
+            ).name
+            self.comment_type = self.store.add_to_comment_types("test_type", self.change_id)
+            self.privacy = self.store.add_to_privacies("test_privacy", 0, self.change_id)
+            self.privacy_id = self.privacy.privacy_id
+            self.privacy = self.privacy.name
+            self.file = self.store.get_datafile(
+                "test_file", "csv", 0, "HASHED", self.change_id, self.privacy
+            )
+            self.file_id = self.file.datafile_id
+            self.file.measurements[self.parser_name] = dict()
+            self.file_2 = self.store.get_datafile(
+                "test_file_2", "csv", 100, "HASHED", self.change_id, self.privacy
+            )
+            self.file_2_id = self.file_2.datafile_id
+            self.platform = self.store.get_platform(
+                platform_name="Test Platform",
+                nationality=self.nationality,
+                platform_type=self.platform_type,
+                privacy=self.privacy,
+                change_id=self.change_id,
+            )
+            self.platform_id = self.platform.platform_id
+            self.sensor = self.platform.get_sensor(
+                self.store, "gps", self.sensor_type, change_id=self.change_id, privacy=self.privacy
+            )
+            self.sensor_id = self.sensor.sensor_id
+            self.state = self.file.create_state(
+                self.store,
+                self.platform,
+                self.sensor,
+                self.current_time,
+                parser_name=self.parser_name,
+            )
+            self.state_2 = self.file.create_state(
+                self.store,
+                self.platform,
+                self.sensor,
+                self.current_time + timedelta(seconds=5),
+                parser_name=self.parser_name,
+            )
+            if self.file.validate():
+                self.file.commit(self.store, self.change_id)
+
+    def test_delete_objects_privacy(self):
+        with self.store.session_scope():
+            result = self.store.find_dependent_objects(constants.PRIVACY, [self.privacy_id])
+            assert "Privacies" in result and result["Privacies"] == 1  # self.privacy
+            assert "Datafiles" in result and result["Datafiles"] == 2  # self.file and self.file_2
+            assert "Platforms" in result and result["Platforms"] == 1  # self.platform
+            assert "Sensors" in result and result["Sensors"] == 1  # self.sensor
+            assert "States" in result and result["States"] == 2  # self.state and self.state_2
+
+            self.store.delete_objects(constants.PRIVACY, [self.privacy_id])
+            result = self.store.find_dependent_objects(constants.PRIVACY, [self.privacy_id])
```

Comment:
Could you add a simple check here to ensure that the privacy object has actually been deleted, by querying for a privacy entry with that ID. I know it should appear in `find_dependent_objects` if it hasn't been deleted, but checking it explicitly would mean we wouldn't be caught out if there was a bug in `find_dependent_objects`.

Reply:
Added: https://github.com/debrief/pepys-import/commit/c766544a7abc49a54b98e8d8a355684d8ae850d1